### PR TITLE
Changes the spares location, Display case code

### PR DESCRIPTION
**Pull request by LadyFowls**
Makes the display case code more modular for other uses by implementing a way to safely open it and a security function for if it is broken in to. Places the new spare id case in the command bunker.

### CODE REVIEW

**Comment by LadyFowls** [comment]
!review

**Comment by Geevies** [comment]
Is there anyway to randomize its location?

**Comment by NonQueueingMatt** [comment]
> 
> 
> Is there anyway to randomize its location?

Possible, but that's a terrible choice imo.

**Comment by NonQueueingMatt** [comment]
Travis fail.

**Comment by NonQueueingMatt** [comment]
nvm lol thanks derelicts

**Comment by Sindorman** [comment]
> Is there anyway to randomize its location?

It is definitely possible to randomize spare location, but should it be random?

File: code/game/objects/structures/displaycase.dm [hunk 1]
```diff
@@ -28,17 +54,34 @@
...
+/obj/structure/displaycase/proc/tripalarm(security_level)
+	var/area/case_area = get_area(src)
+	switch(security_level)
+		if(1)
+			visible_message("<span class='notice'>[src] pings cheerfully.</span>", "<span class='notice'>You hear a ping.</span>")
+		if(2)
+			priority_announcement.Announce("Warning, a company asset has been removed from it's assigned storage case at [case_area.name]. Company assets require command approval to remove from protected cases.", new_sound = 'sound/ai/caseseclevel1.ogg')
+		if(3)
+			set_security_level(SEC_LEVEL_BLUE)
+			priority_announcement.Announce("Warning, a company asset has been removed from the storage case in the [case_area.name]. A blue alert level has been enacted to allow security to recover the asset. All crew are advised to not impede security's inquiry into the assets whereabouts.", new_sound = 'sound/ai/caseseclevel2.ogg')
```
Comment (Sindorman) [inline code/game/objects/structures/displaycase.dm]: This does not look the same as the like above.

`Warning, a company asset has been removed from it's assigned storage case at [case_area.name]`

and 

`Warning, a company asset has been removed from the storage case in the [case_area.name]`

I would say pick either

File: code/game/objects/structures/displaycase.dm [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
+	var/obj/item/displayed_item = null
+	var/held_item = /obj/item/gun/energy/captain
+	var/alarm = TRUE
+	var/access_needed = access_keycard_auth
+	var/open = FALSE
+
+
+/obj/structure/displaycase/spareid
+	name = "spare id containment case"
+	desc = "A display case for holding the spare id, it seems to have a device installed into the side of it."
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]:
```suggestion
	desc = "A display case for holding the captain's spare ID, it seems to have a device installed on the side."
```

File: code/game/objects/structures/displaycase.dm [hunk 3]
```diff
@@ -1,22 +1,47 @@
...
+	held_item = /obj/item/card/id/captains_spare
+	alarm_delay = 30
+	security_level = 2
+
+
+/obj/structure/displaycase/Initialize()
+	. = ..()
+	if(!displayed_item)
+		displayed_item = new held_item(src)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: What's the purpose of this? Why not just use the held item?

File: code/game/objects/structures/displaycase.dm [hunk 4]
```diff
@@ -1,22 +1,47 @@
...
+	var/access_needed = access_keycard_auth
+	var/open = FALSE
+
+
+/obj/structure/displaycase/spareid
+	name = "spare id containment case"
+	desc = "A display case for holding the spare id, it seems to have a device installed into the side of it."
+	held_item = /obj/item/card/id/captains_spare
+	alarm_delay = 30
+	security_level = 2
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: Pretty sure these are defines.

File: code/game/objects/structures/displaycase.dm [hunk 5]
```diff
@@ -1,22 +1,47 @@
...
+	var/alarm = TRUE
+	var/access_needed = access_keycard_auth
+	var/open = FALSE
+
+
+/obj/structure/displaycase/spareid
+	name = "spare id containment case"
+	desc = "A display case for holding the spare id, it seems to have a device installed into the side of it."
+	held_item = /obj/item/card/id/captains_spare
+	alarm_delay = 30
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]:
```suggestion
	alarm_delay = 3 SECONDS
```

File: code/game/objects/structures/displaycase.dm [hunk 6]
```diff
@@ -28,17 +53,34 @@
...
+/obj/structure/displaycase/proc/tripalarm(security_level)
+	var/area/case_area = get_area(src)
+	switch(security_level)
+		if(1)
+			visible_message("<span class='notice'>[src] pings cheerfully.</span>", "<span class='notice'>You hear a ping.</span>")
+		if(2)
+			priority_announcement.Announce("Warning, a company asset has been removed from it's storage case. Company assets require command approval to remove from protected cases.", new_sound = 'sound/ai/caseseclevel1.ogg')
+		if(3)
+			set_security_level(SEC_LEVEL_BLUE)
+			priority_announcement.Announce("Warning, a company asset has been removed from the storage case in the [case_area.name]. A blue alert level has been enacted to allow security to recover the asset. All crew are advised to not impede security's inquiry into the assets whereabouts.", new_sound = 'sound/ai/caseseclevel2.ogg')
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: This is way too long. Please shorten it.

File: code/game/objects/structures/displaycase.dm [hunk 7]
```diff
@@ -28,17 +53,34 @@
...
 	src.healthcheck()
 	return
 
+/obj/structure/displaycase/proc/tripalarm(security_level)
+	var/area/case_area = get_area(src)
+	switch(security_level)
+		if(1)
+			visible_message("<span class='notice'>[src] pings cheerfully.</span>", "<span class='notice'>You hear a ping.</span>")
+		if(2)
+			priority_announcement.Announce("Warning, a company asset has been removed from it's storage case. Company assets require command approval to remove from protected cases.", new_sound = 'sound/ai/caseseclevel1.ogg')
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: This sentence flows really weirdly. Shorten it and make it more succint.

File: code/game/objects/structures/displaycase.dm [hunk 8]
```diff
@@ -47,27 +89,65 @@
...
+	if(istype(W, /obj/item/stack/material) && W.get_material_name() == "glass")
+		var/obj/item/stack/G = W
+		if(health >= maxhealth && !destroyed)
+			to_chat(user, "<span class='notice'>The case cannot be repaired further!.</span>")
+			return
+		if(health <= maxhealth && !destroyed)
+			if(G.use(2))
+				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+				to_chat(user, "<span class='notice'>You places the glass on the [src].</span>")
+				health = max(health+(maxhealth/5), maxhealth)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: You should be using clamp() for this.

File: code/game/objects/structures/displaycase.dm [hunk 9]
```diff
@@ -47,27 +89,65 @@
...
+			update_icon()
+	if(istype(W, /obj/item/stack/material) && W.get_material_name() == "glass")
+		var/obj/item/stack/G = W
+		if(health >= maxhealth && !destroyed)
+			to_chat(user, "<span class='notice'>The case cannot be repaired further!.</span>")
+			return
+		if(health <= maxhealth && !destroyed)
+			if(G.use(2))
+				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+				to_chat(user, "<span class='notice'>You places the glass on the [src].</span>")
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]:
```suggestion
				to_chat(user, "<span class='notice'>You carefully apply some panes and fix the glass on [src].</span>")
```

File: code/game/objects/structures/displaycase.dm [hunk 10]
```diff
@@ -47,27 +89,65 @@
...
+				health = max(health+(maxhealth/5), maxhealth)
+			else
+				to_chat(user, "<span class='warning'>You need 2 sheets of glass to repair this!.</span>")
+	else if(user.a_intent == I_HURT)
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+		src.health -= W.force
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: It isn't wise to do this. This allows you get negative values, which is eh.

File: code/game/objects/structures/displaycase.dm [hunk 11]
```diff
@@ -47,27 +89,65 @@
...
-		src.occupied = 0
+	if (src.open && displayed_item)
+		to_chat(user, "<span class='notice'>You remove the [displayed_item.name].</span>")
+		displayed_item.forceMove(src.loc)
+		displayed_item = null
 		src.add_fingerprint(user)
 		update_icon()
+		if(alarm)
+			if(delayed_alarm)
+				addtimer(CALLBACK(src, .proc/tripalarm, security_level), alarm_delay SECONDS)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: Yeah, uh, don't put the SECONDS there. It should just be alarm_delay. And the seconds should be defined in the alarm_delay declaration.

File: code/game/objects/structures/displaycase.dm [hunk 12]
```diff
@@ -1,22 +1,47 @@
...
 	var/health = 30
-	var/occupied = 1
-	var/destroyed = 0
+	var/maxhealth = 30
+	var/destroyed = FALSE
+	var/item_x_offset = 0
+	var/item_y_offset = 1
+	var/case_security = 1
+	var/delayed_alarm = FALSE
+	var/alarm_delay = 20
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: This is in deciseconds here, which is inconsistent.

File: code/game/objects/structures/displaycase.dm (hunk 3 above)
Comment (NonQueueingMatt) [reply code/game/objects/structures/displaycase.dm]: Redundant variables, still not addressed.

File: code/game/objects/structures/displaycase.dm [hunk 13]
```diff
@@ -28,17 +53,34 @@
...
 	return
 
+/obj/structure/displaycase/proc/tripalarm(case_security)
+	var/area/case_area = get_area(src)
+	switch(case_security)
+		if(1)
+			visible_message("<span class='notice'>[src] pings cheerfully.</span>", "<span class='notice'>You hear a ping.</span>")
+		if(2)
+			priority_announcement.Announce("Warning, a company asset has been removed from it's storage case. Crew are advised to keep assets in there assigned storage cases.")
+		if(3)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]: These should REALLY be defines.

File: code/game/objects/structures/displaycase.dm [hunk 14]
```diff
@@ -28,17 +53,34 @@
...
 	src.healthcheck()
 	return
 
+/obj/structure/displaycase/proc/tripalarm(case_security)
+	var/area/case_area = get_area(src)
+	switch(case_security)
+		if(1)
+			visible_message("<span class='notice'>[src] pings cheerfully.</span>", "<span class='notice'>You hear a ping.</span>")
+		if(2)
+			priority_announcement.Announce("Warning, a company asset has been removed from it's storage case. Crew are advised to keep assets in there assigned storage cases.")
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/displaycase.dm]:
```suggestion
			priority_announcement.Announce("A company asset has been removed from its storage case.")
```
This sentence flows really weirdly. It's not really crew that should be concerned with precious assets, so they shouldn't really be told by the announcement. It should be more like an unspecific warning or an alarm sound that's played.

**Comment by NonQueueingMatt** [comment]
Unaddressed review, conflicts, inactive author. Reopen when ready.

**Pull request closed without merging**